machine/models/baseModel.py

```diff
@@ -0,0 +1,49 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+class Model(nn.Module):
+    """ 
+    Abstract base class for models.
+
+    Args:
+        encoder_module (baseRNN) :  module that encodes inputs
+        decoder_module (baseRNN, optional):   module to decode encoded inputs
+        decode_function (func, optional): function to generate symbols from output hidden states (default: F.log_softmax)
+
+    """
+
+    def __init__(self, encoder_module, decoder_module=None, decode_function=F.log_softmax):
+        super(Model, self).__init__()
+        self.encoder_module = encoder_module
+        self.decoder_module = decoder_module
+        self.decode_function = decode_function
+
+    def flatten_parameters(self):
+        self.encoder_module.rnn.flatten_parameters()
+        self.decoder_module.rnn.flatten_parameters()
+
+    def forward(self, inputs, input_lengths=None, targets=None,
+                teacher_forcing_ratio=0):
+        """
+        Inputs: inputs, input_lengths, targets, teacher_forcing_ratio
+            - **inputs** (list, option): list of sequences, whose length is the batch size and within which
+              each sequence is a list of token IDs. This information is passed to the encoder module.
+            - **input_lengths** (list of int, optional): A list that contains the lengths of sequences
+                in the mini-batch, it must be provided when using variable length RNN (default: `None`)
```

Comment:
Do we want to continue maintaining no variable length RNN ? Doesn't everyone uses those ? If so I think we should remove it.

Reply:
I don't understand the question. Is it related to the baseRNN implementation or something more generic? If it is the latter, could you open an issue about it or a feature request?